runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java

```diff
@@ -157,12 +183,15 @@ public NotificationDto process(File runDirectory, TimeZone tz) throws IOExceptio
       throw new IOException(e);
     }
 
-    // See if we can figure out the chemistry
+    Document runParams = Stream.of("runParameters.xml", "RunParameters.xml").map(f -> new File(runDirectory, f))
+        .filter(file -> file.exists() && file.canRead()).findAny().flatMap(RunProcessor::parseXml).orElse(null);
+    if (runParams != null) {
+      // See if we can figure out the chemistry
+      Arrays.stream(IlluminaChemistry.values()).filter(chemistry -> chemistry.test(runParams)).findFirst()
+          .orElse(IlluminaChemistry.UNKNOWN);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The return value of "orElse" must be used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2201)